### ISSUE TITLE
scrollReachesRightEnd should be recalculated in checkNavStatus() and Window Resize Support

### ISSUE THREAD

**Issue by jayrungta** [bfwg/ngx-drag-scroll#184]
* **I'm submitting a ...**
  - [ X ] bug report
  - [ X ] feature request
  - [ ] support request => Please do not submit support request here, see note at the top of this template.

In my application, I am listening to the window resize event, and subsequently calling the checkNavStatus() function in hopes that it will correctly update the values of reachesLeftBound and reachesRightBound.

However, it fails in recalculating the scrollReachesRightEnd and uses the stale value from the last time onScrollHandler() was called.

I would also like to suggest that you add support for window resize as a feature! It would be as simple as listening to the window resize event and calling the checkNavStatus() function (after this bug is fixed).

**Comment by jayrungta** [comment]
@bfwg thanks for the dynamic checking update. 

Do you think using resize listener with a debounce mechanism is also not recommended? Here's what I am doing:

```
 window.addEventListener('resize', (event) => {
      if (this.ds) {
        if (timer) {
          clearInterval(timer);
        }
        timer = setTimeout(() => {
          this.ds.checkNavStatus();
        }, 200);
      }
    });
```

**Comment by bfwg** [comment]
Debounce is definitely better to have when using the window resize listener in my opinion.

**Comment by bfwg** [comment]
Yes, but I don't have time to do that right now. I can accept a PR if you want to make one.
